### PR TITLE
Snowflake: better error messages, support for role and long queries

### PR DESCRIPTION
**Pull request by sgrebnov**
1. Move `account` identifier from dataset definition to secrets as it is user account specific.
2. Support for `account` identifier with '<orgname.account_name>' format. Default when your copy account identifier from Snowflake account info UI
3. Make `warehouse` parameter optional (Snowflake supports defining default Warehouse, for example for user or role) and make naming consistent with other data connectors: `snowflake_warehouse`
4. Add `snowflake_role` parameter to define target role to use
5. Better error messages and support for longer Snowflake queries by switching to custom swnoflake-api branch (0.8.0-async-exec).  Corresponding PRs: 
- https://github.com/mycelial/snowflake-rs/pull/43
- https://github.com/mycelial/snowflake-rs/pull/42

### CODE REVIEW

**Review by phillipleblanc — DISMISSED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -60,4 +60,4 @@ uuid = "1.6.1"
 pem = "3.0.4"
 odbc-api = { version = "7.0.0" }
 arrow-odbc = { version = "9.0.0" }
-snowflake-api = { version = "0.8.0"}
+snowflake-api = { git="https://github.com/sgrebnov/snowflake-rs.git", folder="snowflake-api", branch="0.8.0-async-exec" }
```
Comment (phillipleblanc) [inline Cargo.toml]: Let's add these commits to https://github.com/spiceai/snowflake-rs

File: crates/db_connection_pool/src/snowflakepool.rs [hunk 2]
```diff
@@ -81,19 +78,21 @@ impl SnowflakeConnectionPool {
...
         let account = get_param(params, secret, "account")
-            .context(MissingRequiredParameterSnafu { name: "account" })?;
-        let warehouse = get_param(params, secret, "warehouse")
-            .context(MissingRequiredParameterSnafu { name: "warehouse" })?;
+            .context(MissingRequiredSecretSnafu { name: "account" })?;
+        let warehouse = get_param(params, secret, "snowflake_warehouse");
+        let role = get_param(params, secret, "snowflake_role");
+
+        // account identifier can be in <orgname.account_name> format but API requires it as <orgname-account_name>
+        let account = account.replace('.', "-");
```
Comment (phillipleblanc) [inline crates/db_connection_pool/src/snowflakepool.rs:87]: Can org/account names have periods `.`?

Comment (sgrebnov) [reply crates/db_connection_pool/src/snowflakepool.rs:87]: Confirm both org name and user name can NOT contain '.'
image

File: Cargo.toml (hunk 1 above)
Comment (sgrebnov) [reply Cargo.toml]: Updated - thank you

**Review by phillipleblanc — DISMISSED** [review]

**Review by digadeesh — APPROVED** [review]

**Pull request merged**